### ISSUE TITLE
Package 'memflow-kvm' was not found in 'dev' channel via 'binary' installation.

### ISSUE THREAD

**Issue by iHackerJi** [memflow/memflowup#5]
I can't install memflow-kvm plugn form memflowup

input : memflowup install -d memflow-kvm
out:
Package 'memflow-kvm' was not found in 'dev' channel via 'binary' installation.
'memflow-kvm' is only available via the following channels: dev/git
Use 'memflowup install --help' to see additional flags for choosing a channel.

**Comment by Blightbuster** [comment]
You also need to tell it to install from source (Use the help command to see the flag for it).

I agree that it's kind of confusing right now.

**Comment by iHackerJi** [comment]
> 

thank you guy,now i can download and build  it,
but when I build memflowup-kvm.it faild,eror input:

error: failed to run custom build command for `memflow-kvm-ioctl v0.1.0 (/tmp/memflowup_build/11956959929559675329/b07ac5e5ea12266a/memflow-kvm-ioctl)`

Caused by:
  process didn't exit successfully: `/tmp/memflowup_build/11956959929559675329/b07ac5e5ea12266a/target/release/build/memflow-kvm-ioctl-e359d2e12f335484/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=../mabi.h
  cargo:rerun-if-changed=wrapper.h

  --- stderr
  thread 'main' panicked at 'Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"', /home/zdj/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-7a04d2510079875b/bindgen-0.58.1/src/lib.rs:2057:31
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
Error: "Runtime error: failed to open artifact (line 6, position 5) in call to function build_from_path < build_from_source"


have you encountered this problem？

**Comment by Blightbuster** [comment]
You'll have to install libclang as it tells you to.

**Comment by ko1N** [comment]
We should add some preflight checks for existing packages and tools and give a more explicit error message in case those are missing.

**Comment by iHackerJi** [comment]
> You'll have to install libclang as it tells you to.

ok，now i can use memflow library.thanks for your explain

**Comment by iHackerJi** [comment]
> We should add some preflight checks for existing packages and tools and give a more explicit error message in case those are missing.

yes man,when i use memflowup,I didn't expect have a option like: memflowup install -h，i think just have a way to get help info —— "memflowup -h"。

**Comment by Tacodiva** [comment]
Just FYI, this command:
`memflowup install memflow-kvm -s -d --reinstall`
is in the README but from my understanding that will never work because you don't specify `-S`.

**Comment by ko1N** [comment]
Since this issue is related to memflowup 0.1 ill close it
